### PR TITLE
feat: support --replace|merge with conf load

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes https://emqx.atlassian.net/browse/EMQX-10263

### <samp>🤖 Generated by Copilot at ae0ab75</samp>

Enhanced the `conf load` command with new options to replace or merge the current configuration with a new file. Improved the error handling and feedback of the configuration operations in `emqx_conf_cli.erl`.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_conf/src/emqx_conf_cli.erl [hunk 1]
```diff
@@ -98,17 +102,21 @@ admins(_) ->
...
-            "NOTE: do not make runtime config changes during rolling upgrade."}
+            "Print in-use configs (including default values) under the given key."},
+        {"", "Print ALL keys if key is not provided"},
+        {"conf load --replace|--merge <path>", "Load a HOCON format config file."},
+        {"", "The new configuration values will be overlaid on the existing values by default."},
+        {"", "use the --replace flag to replace existing values with the new ones instead."},
+        {"", "The current node will initiate a cluster wide config change"},
+        {"", "transaction to sync the changes to other nodes in the cluster. "},
+        {"", "NOTE: do not make runtime config changes during rolling upgrade."},
+        {"----------------------------------", "------------"}
```
Comment (zhongwencool) [inline apps/emqx_conf/src/emqx_conf_cli.erl:119]:
```
./bin/emqx_ctl conf
conf reload --replace|--merge                  # reload etc/emqx.conf on local node
                                               # The new configuration values will be overlaid on the existing values by default.
                                               # use the --replace flag to replace existing values with the new ones instead.
----------------------------------             # ------------
conf show_keys                                 # print all the currently used configuration keys.
conf show [<key>]                              # Print in-use configs (including default values) under the given key.
                                               # Print ALL keys if key is not provided
conf load --replace|--merge <path>             # Load a HOCON format config file.
                                               # The new configuration values will be overlaid on the existing values by default.
                                               # use the --replace flag to replace existing values with the new ones instead.
                                               # The current node will initiate a cluster wide config change
                                               # transaction to sync the changes to other nodes in the cluster.
                                               # NOTE: do not make runtime config changes during rolling upgrade.
----------------------------------             # ------------
conf cluster_sync status                       # Show cluster config sync status summary
conf cluster_sync skip [node]                  # Increase one commit on specific node
conf cluster_sync tnxid <TnxId>                # Display detailed information of the config change transaction at TnxId
conf cluster_sync fast_forward [node] [tnx_id] # Fast-forward config change transaction to tnx_id on the given node.WARNING: This results in inconsistent configs among the clustered nodes.
```

File: apps/emqx_conf/src/emqx_conf_cli.erl [hunk 2]
```diff
@@ -162,19 +170,7 @@ drop_hidden_roots(Conf) ->
     maps:without(Hidden, Conf).
 
 hidden_roots() ->
-    SchemaModule = emqx_conf:schema_module(),
```
Comment (zhongwencool) [inline apps/emqx_conf/src/emqx_conf_cli.erl:165]: some hide root key also need update, e.g rule_engine topic_rewrite...

File: apps/emqx_conf/src/emqx_conf_cli.erl [hunk 3]
```diff
@@ -216,15 +212,34 @@ load_config(Path, AuthChain) ->
             {error, bad_hocon_file}
     end.
 
-update_config(?EMQX_AUTHORIZATION_CONFIG_ROOT_NAME = Key, Conf, "merge") ->
+update_config_cluster(?EMQX_AUTHORIZATION_CONFIG_ROOT_NAME = Key, Conf, merge) ->
     check_res(Key, emqx_authz:merge(Conf));
-update_config(?EMQX_AUTHENTICATION_CONFIG_ROOT_NAME = Key, Conf, "merge") ->
+update_config_cluster(?EMQX_AUTHENTICATION_CONFIG_ROOT_NAME = Key, Conf, merge) ->
```
Comment (zhongwencool) [inline apps/emqx_conf/src/emqx_conf_cli.erl:217]: authz/autn is a list, they have special merge rule internal.

File: apps/emqx_conf/src/emqx_conf_cli.erl [hunk 4]
```diff
@@ -290,23 +309,13 @@ load_etc_config_file() ->
             {error, Error}
     end.
 
-check_readonly_config(Raw) ->
+filter_readonly_config(Raw) ->
     SchemaMod = emqx_conf:schema_module(),
     RawDefault = emqx_config:fill_defaults(Raw),
     case emqx_conf:check_config(SchemaMod, RawDefault) of
-        {ok, CheckedConf} ->
-            case filter_changed_readonly_keys(CheckedConf) of
```
Comment (zhongwencool) [inline apps/emqx_conf/src/emqx_conf_cli.erl:298]: before: 
the check will failed if  readonly key changed.
after: 
we just filter readonly key.

**Review by HJianBo — APPROVED** [review]

**Pull request merged**